### PR TITLE
storageClass should be set if mancenter persistence is enabled

### PR DESCRIPTION
**Pull request by jisnardo**
storageClass should be set if mancenter persistence is enabled

```
[jisnardo@ed1624 hazelcast]$ k8s -n hazelcast3 get pods
NAME                    READY   STATUS    RESTARTS   AGE
hazelcast-0             1/1     Running   0          123m
hazelcast-1             1/1     Running   0          122m
hazelcast-2             1/1     Running   0          122m
hazelcast-mancenter-0   0/1     Pending   0          123m

[jisnardo@ed1624 ~]$ k8s -n hazelcast3 describe pvc/hazelcast-mancenter
Name:          hazelcast-mancenter
Namespace:     hazelcast3
StorageClass:  
Status:        Pending
Volume:        
Labels:        app.kubernetes.io/instance=hazelcast
               app.kubernetes.io/managed-by=Helm
               app.kubernetes.io/name=hazelcast
               helm.sh/chart=hazelcast-2.10.1
Annotations:   meta.helm.sh/release-name: hazelcast
               meta.helm.sh/release-namespace: hazelcast3
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Mounted By:    hazelcast-mancenter-0
Events:
  Type    Reason         Age                    From                         Message
  ----    ------         ----                   ----                         -------
  Normal  FailedBinding  4m28s (x302 over 79m)  persistentvolume-controller  no persistent volumes available for this claim and no storage class is set
```

### CODE REVIEW

**Comment by leszko** [comment]
Why do you want to set the storage class to `my-storage-class` by default?

**Comment by jisnardo** [comment]
```
  persistence:
    # enabled is a flag to enable persistence for Management Center
    enabled: true
```

requires an storageClass for avoid no persistent volumes available for this claim and no storage class is set.

my-storage-class aims to be any one of available.

https://kubernetes.io/docs/concepts/storage/storage-classes/

Example:

storageClass: csi-rbd-sc

```
[jisnardo@pc ~]$ k8s get sc
NAME           PROVISIONER         AGE
ceph-com-rbd   ceph.com/rbd        69d
ceph-rbd       kubernetes.io/rbd   104d
csi-rbd-sc     rbd.csi.ceph.com    54d
```

so storageClass: whatever your cluster have

or set  by default

```
  persistence:
    # enabled is a flag to enable persistence for Management Center
    enabled: false
```

**Comment by leszko** [comment]
I don't think we want to have any storageClass defined by default. Could you try on some GKE, Minikube, or OpenShift Online environments? Maybe it only fails in your environment.

**Comment by jisnardo** [comment]
We use on-premise kubernetes with ceph csi-rbd, so this could be the clue, with default values.yaml persistentvolume-controller complains because mancenter persistence is enabled.

So it would be enough adding the proposal as comment suggestion:

```
    # https://kubernetes.io/docs/concepts/storage/storage-classes/
    #storageClass: my-storage-class
```

**Comment by leszko** [comment]
Yes, you're right, we should add a commented-out storageClass to `values.yaml`. Could you update your PR to include only this change?

Thanks!

**Comment by jisnardo** [comment]
PR updated.

```
+    # https://kubernetes.io/docs/concepts/storage/storage-classes/
+    #storageClass: my-storage-class
```

Thank you too.

File: stable/hazelcast-enterprise/values.yaml [hunk 1]
```diff
@@ -330,6 +330,8 @@ mancenter:
     - ReadWriteOnce
     # size is the size of Persistent Volume Claim
     size: 8Gi
+    # https://kubernetes.io/docs/concepts/storage/storage-classes/
+    #storageClass: my-storage-class
```
Comment (leszko) [inline stable/hazelcast-enterprise/values.yaml:334]:
```suggestion
    # storageClass defines the storage class used for Management Center. Read more at: https://kubernetes.io/docs/concepts/storage/storage-classes/
    # storageClass:
```

File: stable/hazelcast/values.yaml [hunk 2]
```diff
@@ -296,6 +296,8 @@ mancenter:
     - ReadWriteOnce
     # size is the size of Persistent Volume Claim
     size: 8Gi
+    # https://kubernetes.io/docs/concepts/storage/storage-classes/
+    #storageClass: my-storage-class
```
Comment (leszko) [inline stable/hazelcast/values.yaml:300]:
```suggestion
    # storageClass defines the storage class used for Management Center. Read more at: https://kubernetes.io/docs/concepts/storage/storage-classes/
    #storageClass:
```

**Review by leszko — COMMENTED** [review]
Thanks a lot @jisnardo 

Added two suggestions just to keep consistency with other parts of `values.yaml`.

Thanks again for raising this PR.

**Review by leszko — COMMENTED** [review]
@jisnardo did you apply the changes I suggested?

**Comment by hasancelik** [comment]
verify

**Comment by jisnardo** [comment]
Sorry @leszko i forgot it, done just now, thanks

https://github.com/jisnardo/charts/commit/628991de356eccbcf171cdbe68ba8a46d9358f6b

**Comment by hasancelik** [comment]
Hi @jisnardo, thanks for your contribution 👍 

PR builder failed at linting step. Could you rebase your PR and bump up chart versions?
```
11:29:34 Linting chart 'hazelcast-enterprise => (version: "3.4.11", path: "stable/hazelcast-enterprise")'
11:29:34 Checking chart 'hazelcast-enterprise => (version: "3.4.11", path: "stable/hazelcast-enterprise")' for a version bump...
11:29:34 Old chart version: 3.4.11
11:29:34 New chart version: 3.4.11
11:29:34 Linting chart 'hazelcast => (version: "3.4.10", path: "stable/hazelcast")'
11:29:34 Checking chart 'hazelcast => (version: "3.4.10", path: "stable/hazelcast")' for a version bump...
11:29:34 Old chart version: 3.4.10
11:29:34 New chart version: 3.4.10
11:29:34 Error: Error linting charts: Error processing charts
```

File: stable/hazelcast-enterprise/values.yaml [hunk 3]
```diff
@@ -330,6 +330,7 @@ mancenter:
     - ReadWriteOnce
     # size is the size of Persistent Volume Claim
     size: 8Gi
+    # storageClass defines the storage class used for Management Center. Read more at: https://kubernetes.io/docs/concepts/storage/storage-classes/
```
Comment (leszko) [inline stable/hazelcast-enterprise/values.yaml:333]: Could you also add `# storageClass:` as I suggested here?

image

**Comment by jisnardo** [comment]
I think i've fixed.

**Comment by hasancelik** [comment]
@jisnardo Chart version issues are resolved. But as @leszko mentioned/suggested above, adding commented out `storageClass:` parameter into `values.yaml` files would be good to help other users:
```
    # storageClass defines the storage class used for Management Center. Read more at: https://kubernetes.io/docs/concepts/storage/storage-classes/
    #storageClass:
```

**Comment by jisnardo** [comment]
I was pushing to another backup repo, just solved.

Thanks @leszko @hasancelik.

**Review by hasancelik — APPROVED** [review]

**Review by leszko — APPROVED** [review]

**Pull request merged**